cardinal/system/t_meat_puppet_system_test.go

```diff
@@ -0,0 +1,290 @@
+			output, move, err := ProcessCommandsTokensLogicTest(tt.tokens, tt.player.PlayerID, tt.player.RoomID)
+
+			assert.Equal(t, tt.expectedOut, output)
+			assert.Equal(t, tt.expectedMove, move)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleAlias tests the handleAlias function
+func TestHandleAlias(t *testing.T) {
+	setup()
+	tests := []struct {
+		name        string
+		tokens      []string
+		playerID    uint32
+		expectedErr uint8
+	}{
+		{
+			name:        "Inventory alias",
+			tokens:      []string{"INVENTORY"},
+			playerID:    6,
+			expectedErr: 135,
+		},
+		{
+			name:        "Look alias",
+			tokens:      []string{"LOOK", "UP"},
+			playerID:    6,
+			expectedErr: 130,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := handleAliasTest(tt.tokens, tt.playerID)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleVerb tests the handleVerb function
+func TestHandleVerb(t *testing.T) {
+	setup()
+	tests := []struct {
+		name        string
+		tokens      []string
+		roomID      uint32
+		playerID    uint32
+		expectedOut string
+		expectedErr uint8
+	}{
+		{
+			name:        "Look verb",
+			tokens:      []string{"LOOK", "TO", "THE", "WINDOW"},
+			roomID:      0,
+			playerID:    6,
+			expectedOut: "---->HANDLE VERB: NOW SHOULD BE GOING TO STUFF FROM LOOK SYSTEM",
+			expectedErr: 0,
+		},
+		{
+			name:        "Take verb",
+			tokens:      []string{"TAKE", "THE", "BALL"},
+			roomID:      0,
+			playerID:    6,
+			expectedOut: "---->HANDLE VERB: NOW SHOULD BE GOING TO TAKE FROM INVENTORY SYSTEM",
+			expectedErr: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output, err := handleVerbTest(tt.tokens, tt.roomID, tt.playerID)
+			assert.Equal(t, tt.expectedOut, output)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestInsultMeat tests the insultMeat function
+func TestInsultMeat(t *testing.T) {
+	tests := []struct {
+		name        string
+		cErr        uint8
+		badCmd      string
+		expectedMsg string
+	}{
+		{
+			name:        "Too many tokens",
+			cErr:        constants.ErrParserRoutineTKCX.Code,
+			badCmd:      "",
+			expectedMsg: "WTF, slow down cowboy, you're gonna hurt yourself",
+		},
+		{
+			name:        "Gibberish command",
+			cErr:        constants.ErrParserRoutineNOP.Code,
+			badCmd:      "",
+			expectedMsg: "Nope, gibberish. Stop breathing with your mouth.",
+		},
+		{
+			name:        "No direction",
+			cErr:        constants.ErrParserRoutineND.Code,
+			badCmd:      "north",
+			expectedMsg: "Go where pilgrim?",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := insultMeatTest(tt.cErr, tt.badCmd)
+			assert.Equal(t, tt.expectedMsg, msg)
+		})
+	}
+}
+
+// Process the Commands tokens, this is the function dedicated to it.
+func ProcessCommandsTokensLogicTest(Tokens []string, PlayerID uint32, roomID uint32) (string, bool, uint8) {
```

Comment:
Im not really sure whats going on here, the tests should be on the actual system. It seems like the tests are in fact testing tests, in that in this test you call the handleAliasTest and the handleVerbTest.

Reply:
Leaving this file until get response from Argus guys